### DISCUSSION TITLE
How can I set it to use the brightness keys on my keyboard?

### DISCUSSION THREAD

**Discussion by mapleroyal** [xanderfrangos/twinkle-tray#633] [Q&A]
MacOS DDC apps are able to recognize the brightness keys so it's obviously possible. Someone said in another thread that it isn't possible but that's either outdated or misinformed.

**Comment by xanderfrangos** [comment]
Hi @mapleroyal,

The information from here is still correct: https://github.com/xanderfrangos/twinkle-tray/issues/291

Most OEM laptop keyboards don't communicate with Windows in a way that is customizable.

**Comment by mapleroyal** [comment]
What about desktop keyboards? My machine is just a desktop. Several DDC apps on Mac won’t let me use the brightness keys but I found one that does. I suppose windows is different but I always thought you can do anything with software with enough ingenuity. Do you mean, it’s possible but extremely difficult/would require a rootkit, or something like that?

**Comment by xanderfrangos** [comment]
Per that issue, you can use Logitech's software to remap many Logitech keyboards. There's similar software for some other keyboard brands. 

To my understanding, MacOS's keyboards sends all keyboard key presses to the OS, where the OS then decides what to do ("increase brightness" for example). On Windows, only standard keys are sent to the OS. Hardware-specific keys (like on most laptops) do not send key input to Windows. Instead the keyboard sends the "action" to Windows ("increase brightness"). That "action" cannot be hooked/modified/etc, especially when the "action" is interpreted by a driver.

As with anything software related, it is probably possible on a per-keyboard basis with a custom driver, custom firmware, or some wild C++. But that's well outside of the scope of Twinkle Tray and I have no interest in getting involved with that.
